app-5-rdf-serializer/src/test/java/org/geovistory/toolbox/streams/rdf/processors/project/ProjectOwlClassTest.java

```diff
@@ -0,0 +1,228 @@
+
+class ProjectOwlClassTest {
+
+    private static final String SCHEMA_REGISTRY_SCOPE = ProjectOwlClassTest.class.getName();
+    private static final String MOCK_SCHEMA_REGISTRY_URL = "mock://" + SCHEMA_REGISTRY_SCOPE;
+    private TopologyTestDriver testDriver;
+    private TestInputTopic<ProjectClassLabelKey, ProjectClassLabelValue> projectClassLabel;
+    private TestOutputTopic<ProjectRdfKey, ProjectRdfValue> outputTopic;
+
+    @BeforeEach
+    void setup() {
+
+
+        Properties props = new Properties();
+        var appId = "test";
+        props.put(StreamsConfig.APPLICATION_ID_CONFIG, appId);
+        props.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy:1234");
+        props.put(StreamsConfig.STATE_DIR_CONFIG, "/tmp/kafka-streams-test");
+        var builderSingleton = new BuilderSingleton();
+        var avroSerdes = new AvroSerdes();
+        avroSerdes.QUARKUS_KAFKA_STREAMS_SCHEMA_REGISTRY_URL = MOCK_SCHEMA_REGISTRY_URL;
+        var inputTopicNames = new InputTopicNames();
+        var outputTopicNames = new OutputTopicNames();
+        var registerInputTopic = new RegisterInputTopic(avroSerdes, builderSingleton, inputTopicNames);
+        var communityClassLabel = new ProjectOwlClass(avroSerdes, registerInputTopic, outputTopicNames);
+        communityClassLabel.addProcessorsStandalone();
+        var topology = builderSingleton.builder.build();
+        testDriver = new TopologyTestDriver(topology, props);
+
+        projectClassLabel = testDriver.createInputTopic(
+                inputTopicNames.getProjectClassLabel(),
+                avroSerdes.ProjectClassLabelKey().serializer(),
+                avroSerdes.ProjectClassLabelValue().serializer());
+
+        outputTopic = testDriver.createOutputTopic(
+                outputTopicNames.projectRdf(),
+                avroSerdes.ProjectRdfKey().deserializer(),
+                avroSerdes.ProjectRdfValue().deserializer());
+    }
+
+    @AfterEach
+    void teardown() {
+        testDriver.close();
+    }
+
+    /**
+     * The goal of this test is to check if the output is not empty
+     */
+    @Test
+    void testOutputIsNotEmpty() {
+
+        var projectId = 1;
+        var classId = 123212;
+        var label = "test";
+        var lang = "en";
+
+        var k = ProjectClassLabelKey.newBuilder().setProjectId(projectId).setClassId(classId).build();
+        var v = ProjectClassLabelValue.newBuilder()
+                .setProjectId(projectId)
+                .setClassId(classId)
+                .setLabel(label)
+                .setLanguageIso(lang)
+                .setLanguageId(I.EN.get())
+                .setDeleted$1(false)
+                .build();
+        projectClassLabel.pipeInput(k, v);
+
+        assertThat(outputTopic.isEmpty()).isFalse();
+    }
+
+    /**
+     * The goal of this test is to check if the output has the correct size (should be equals to twice the number of statements in the input topic)
```

Comment:
Comment has to be updated:

(should be twice the number of records in the input topic)

Reply:
Done